Jint/Runtime/Interop/MethodInfoFunctionInstance.cs

```diff
@@ -9,19 +9,18 @@ namespace Jint.Runtime.Interop
 {
     internal sealed class MethodInfoFunctionInstance : FunctionInstance
     {
-        private static readonly JsString _name = new JsString("Function");
         private readonly MethodDescriptor[] _methods;
         private readonly ClrFunctionInstance? _fallbackClrFunctionInstance;
 
-        public MethodInfoFunctionInstance(Engine engine, MethodDescriptor[] methods)
-            : base(engine, engine.Realm, _name)
+        public MethodInfoFunctionInstance(Engine engine, MethodDescriptor[] methods, string name)
```

Comment:
nit: name feels more natural to come before methods

Reply:
I will add a `Type targetType` to `MethodInfoFunctionInstance`.
Would you like the arguments sequence?
```c#
public void MakeProperty(
  Engine engine,
  Type targetType,
  object target,
  MemberInfo memberInfo,
  string memberName,
  MethodDescriptor[] methods,
  ClrFunctionInstance? fallbackClrFunctionInstance = null
) {}
```

Reply:
It doesn't matter that much as types are internal, we can always shuffle order later on without breaking API